charts/invenio/values.yaml

```diff
@@ -90,7 +85,7 @@ nginx:
   denied_uas: ""
 
 web:
-  image: your/invenio-image
+  image: "ghcr.io/inveniosoftware/demo-inveniordm/demo-inveniordm"
```

Comment:
Yeah, `latest` should generally be avoided, implicit or not. In my values, I have it pinned to `ghcr.io/inveniosoftware/demo-inveniordm/demo-inveniordm@sha256:2193abc2caec9bc599061d6a5874fd2d7d201f55d1673a545af0a0406690e8a4` but I thought that was too long and precise to be a good default value.

Are there semver tagged images we can use instead?

Reply:
> I don't like the thought of having to remember to bump it every time there's a new RDM version released.

That is generally what one has to do to keep the chart up to date, though. Maybe some CI tool could be used to create the PRs automatically. That's what Bitnami do, I'm pretty sure. Because they do release new versions of their chart everytime there's a new app version released.

A trick I used in a previous project, which was fairly easy to maintain, was to use `.Release.AppVersion` (which comes from the `appVersion` field of the `Chart.yaml` file) as the image tag. That let us achieve a couple of things:

* The `appVersion` indicated by the chart was always the one used.
* We could specify `appVersion` during the `helm package` step in our build pipeline, which substitutes the `appVersion` field with whatever value is specified. So that way, we could have `appVersion: latest` or whatever in the Git repo and let the published Helm charts have pinned `appVersions`.

There's some drawbacks of this model:

* If web and worker ever gets different versions, this release model breaks.
* The version published never exactly corresponds with the version in Git.

Reply:
I've pushed a new `v12.0.0-beta.1.3` tag in the demo-inveniordm repo so we get a new image published at https://github.com/inveniosoftware/demo-inveniordm/pkgs/container/demo-inveniordm%2Fdemo-inveniordm

Not sure how "stable" it would be, but let's set it for now for a start.

Reply:
Good. I've set the tag and pushed the change now.

Reply:
Wait actually, I think I want to try and do a thing with the `appVersion` now that I think about it! Hang on!

Reply:
Sorry, I missed https://github.com/inveniosoftware/helm-invenio/pull/103#discussion_r1490954506 by a minute in my reply.

I like this as an approach, but the case with the chart here is that downstream users will be heavily customizing and building their own images and releasing tags at their own time. The approach with having an `appVersion` would work if we had a "final" product which we heavily control the releases and users don't really customize that much besides some basic config.